testsuite/tests/printing-types/pr248.ocaml.reference

```diff
@@ -1,9 +1,9 @@
 val f : 'a -> 'b -> 'b = <fun>
-val blah : '_weak1 -> '_weak1 = <fun>
-val splash : unit -> '_weak1 = <fun>
-val blurp : '_weak2 -> '_weak2 = <fun>
+val blah : '_a -> '_a = <fun>
+val splash : unit -> '_a = <fun>
```

Comment:
Do you have an idea for why `'_weak1` became `'_a` here? There are no annotations in [the code](https://github.com/ocaml/ocaml/blob/943f653d73a657cdce991d8be96c3d1b137dec5b/testsuite/tests/printing-types/pr248.ml). I find `'_weak*` more explicit as a weak variable name, so I wouldn't mind keeping it. (Also, I don't understand why the other weak variables further in the test keep their `_weak` prefix.)

Reply:
Yes, I looked at this and wondered whether this was an improvement. I decided that the `_` prefix was enough to denote a weak variable... but perhaps that's wrong. If you think this is important, I can delve further. I do think that if a weak variable arises from an annotated definition, it will have only the `_` prefix without the `weak`, so I'm not sure it's necessarily worth the time to understand this particular non-annotated example. On the other hand, if we want a policy of always using `weak`, I can look into that.

Reply:
I think that we started using `weak` specifically to draw user attentions to the issue here (which can create surprises down the road), so yes it would make sense to keep `weak`. Maybe we could use `_weak_a`, that is, use `_weak` as a prefix for the name when it exists, and keep using `_weak<n>` otherwise.

Reply:
The `'a` name comes from the `failwith` function,  this is why the `_'a` name only appears in the presence of `splash`:
```ocaml
let failme: unit -> 'never = fun () -> failwith "id"
let id x = x (* Fun.id has type 'a -> 'a thus would introduce the name 'a *)
let blah = id id
let splash () = blah (failme ())
```
```ocaml
val failme : unit -> 'never = <fun>
val id : 'a -> 'a = <fun>
val blah : '_never -> '_never = <fun>
val splash : unit -> '_never = <fun>
```
This is however an illustration that names are made quite stickier from this change.

Reply:
I've pushed a commit that uses `weak` for all weak variables. I agree this is an improvement.